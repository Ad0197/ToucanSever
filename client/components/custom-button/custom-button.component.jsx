import React from "react";
import { CustomButtonContainer } from "./custom-button.styles";

const CustomButton = ({ children, onClick, color = "primary", rounded = true }) => <CustomButtonContainer {...{ onClick, color, rounded }}>{children}</CustomButtonContainer>;

export default CustomButton;
