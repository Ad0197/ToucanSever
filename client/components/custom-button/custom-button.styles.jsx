import styled from "styled-components";

export const CustomButtonContainer = styled.button`
  background-color: ${({ color, theme }) => theme[color]};
  outline: none;
  border: none;
  cursor: pointer;
  font-weight: 500;
  padding: 15px 30px;
  border-radius: ${({ theme, rounded }) => (rounded ? theme.borderRadiusButton : "0px")};
  font-size: 14px;
  transition: opacity 200ms ease-in, box-shadow 200ms ease;

  &:hover {
    opacity: 0.9;
    box-shadow: 0px 0px 15px 1px rgba(0, 0, 0, 0.3);
  }
`;
