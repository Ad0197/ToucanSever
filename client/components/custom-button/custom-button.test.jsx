import { shallow } from "enzyme";
import { CustomButtonContainer } from "./custom-button.styles";
import CustomButton from "./custom-button.component";

describe("CustomButton components", () => {
  it("should show childrens", () => {
    const text = "TEST";
    const wrapper = shallow(<CustomButton>{text}</CustomButton>);
    expect(wrapper.text()).toBe(text);
  });

  it("should have a onClick Props", () => {
    const onClick = jest.fn();
    const wrapper = shallow(<CustomButton onClick={onClick} />);
    wrapper.simulate("click");
    expect(onClick).toHaveBeenCalled();
  });

  it("should have a rounded props by default true", () => {
    const wrapper = shallow(<CustomButton />);
    expect(wrapper.prop("rounded")).toBeTruthy();
  });

  it("should have a prop color by default primary", () => {
    const wrapper = shallow(<CustomButton />);
    expect(wrapper.prop("color")).toBe("primary");
  });
});
