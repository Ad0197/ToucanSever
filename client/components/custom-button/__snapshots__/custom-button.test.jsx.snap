// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton components should set color when you pass a color in 1`] = `
<custom-buttonstyles__CustomButtonContainer
  color="dark"
  rounded={true}
/>
`;
